rtic-time/src/lib.rs

```diff
@@ -77,25 +77,25 @@ pub struct TimeoutError;
 
 /// This is needed to make the async closure in `delay_until` accept that we "share"
 /// the link possible between threads.
-struct LinkPtr<Mono: Monotonic>(*mut Option<linked_list::Link<WaitingWaker<Mono>>>);
+struct LinkPtr<Instant: Copy + Ord>(*mut Option<linked_list::Link<WaitingWaker<Instant>>>);
 
-impl<Mono: Monotonic> Clone for LinkPtr<Mono> {
+impl<Instant: Copy + Ord> Clone for LinkPtr<Instant> {
     fn clone(&self) -> Self {
         LinkPtr(self.0)
     }
 }
 
-impl<Mono: Monotonic> LinkPtr<Mono> {
+impl<Instant: Copy + Ord> LinkPtr<Instant> {
     /// This will dereference the pointer stored within and give out an `&mut`.
-    unsafe fn get(&mut self) -> &mut Option<linked_list::Link<WaitingWaker<Mono>>> {
+    unsafe fn get(&mut self) -> &mut Option<linked_list::Link<WaitingWaker<Instant>>> {
         &mut *self.0
     }
 }
 
-unsafe impl<Mono: Monotonic> Send for LinkPtr<Mono> {}
-unsafe impl<Mono: Monotonic> Sync for LinkPtr<Mono> {}
+unsafe impl<Instant: Copy + Ord> Send for LinkPtr<Instant> {}
+unsafe impl<Instant: Copy + Ord> Sync for LinkPtr<Instant> {}
 
-impl<Mono: Monotonic> TimerQueue<Mono> {
```

Comment:
I think we should rename `TimerQueue` to `PriorityQueue` and add a type alias similar to `type TimerQueue<T> = PriorityQueue<<T as Monotonic>::Instant>`